rpc/http.go

```diff
@@ -173,6 +173,9 @@ func (c *Client) sendBatchHTTP(ctx context.Context, op *requestOp, msgs []*jsonr
 	if err := json.NewDecoder(respBody).Decode(&respmsgs); err != nil {
 		return err
 	}
+	if len(respmsgs) != len(msgs) {
+		return fmt.Errorf("batch has %d requests but response has %d: %w", len(msgs), len(respmsgs), ErrBadResult)
+	}
```

Comment:
Like, why? Seriously?

Reply:
It has to do with channel issues and locking. 
Check here  https://github.com/ethereum/go-ethereum/pull/26064